src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -1077,6 +1052,13 @@ abstract class BTypes {
       "scala/Null",
       "scala/Nothing"
     )
+
+    def apply(internalName: InternalName) : ClassBType = {
+      classBTypeFromInternalName.getOrElseUpdate(internalName, new ClassBType(internalName))
```

Comment:
We should probably make sure there's a single ClassBType per InternalName, also under concurrent access. I haven't checked in detail if something depends on this assumption.